<!-- -->

<launch>

   <node name="optitrack" pkg="sim_optitrack" type="sim_optitrack_node" output="screen">
			<rosparam>rigid_bodies_name:  [quadrotor]</rosparam>
	</node>

    <node name="set_waypoints_node" pkg="set_waypoints" type="ar_drone_set_waypoints"  output="screen">
	<remap from="waypoint_topic"  to="/drone_target_points"/>
	<remap from="drone_info_topic"  to="/optitrack/quadrotor"/>
	<remap from="waypoint_info_topic"  to="/set_waypoints_info"/>
	<param name="range" value="10.0"/>
	<param name="stop_mission" value="false"/>
	<param name="flight_time" value="50.0"/>
	<param name="emergency" value="false"/>
	<param name="minumum_radius" value="0.2"/>
	<rosparam>central_point:  [5.0, 5.0]</rosparam>  <!-- [x, y] -->
	<rosparam>final_position:  [-5.0, -5.0]</rosparam>  <!-- [x, y] -->
	<rosparam>emergency_position:  [1.0, 1.0]</rosparam>  <!-- [x, y] -->
    </node>

   <node name="drone_control_node" pkg="drone_control_position" type="control_drone_position_node" output="screen">
		<remap from="drone_info_topic"  to="/optitrack/quadrotor"/>
		<remap from="control_info"  to="/drone_target_points"/>
		<rosparam>Kd:  [0.5, 0.5]</rosparam>  <!-- [pitch, roll] -->
		<rosparam>Kp:  [0.5, 0.5, 0.25, 0.05]</rosparam>  <!-- [pitch, roll, altitude, yaw] -->
		<rosparam>velocity_limit:  [1.0, 1.0, 1.0, 1.0]</rosparam>  <!-- [pitch, roll, altitude, yaw] -->
		<rosparam>delta_pose:  [0.0, 0.0, 0.0, 0.0]</rosparam>  <!-- [x,y,z,yaw (degrees)] -->
		<rosparam>virtual_fence:  [20.0, -20.0, 20.0, -20.0, 2.5]</rosparam>  <!-- [maxX, minX, maxY, minY, maxAltitude] -->
	</node>

</launch>
